voting/src/voting.rs

```diff
@@ -216,10 +218,19 @@ impl TxOwner for Context {
                 if choice != VoteChoice::Favor && choice != VoteChoice::Against && choice != VoteChoice::Absention {
                     return Err(ExecuteError::InvalidChoice).map_err(|_| todo_fixthis)
                 }
-                let mut vote_paper: VotePaper =
-                    serde_cbor::from_slice(&self.storage().get(vote_paper_id.0.as_ref()).unwrap())
-                        .map_err(|_| ExecuteError::VotePaperNotFound)
-                        .map_err(|_| todo_fixthis)?;
+
+                if !self.storage().has(vote_paper_id.0.as_ref()) {
+                    return Err(ExecuteError::VotePaperNotFound).map_err(|_| todo_fixthis)
+                }
+
+                let mut vote_paper: VotePaper = {
+                    let bytes = &self
+                        .storage()
+                        .get(vote_paper_id.0.as_ref())
+                        .expect("Previously we checked the existence of the vote paper");
+                    serde_cbor::from_slice(bytes).expect("Vote paper is serialized by this code")
+                };
+
```

Comment:
This looks not related to "Add a missing function to update vote box changes" commit.